File: 10.py
s = 0
primes = []


def is_prime(x, primes):
    for n in primes:
        if n**2 > x:
            break
        if x % n == 0:
            return False
    primes.append(x)
    return True


for i in range(2, int(2e6)):
    if is_prime(i, primes):
        s += i

print(s)

File: JavaScript/10.js
function primesSmallerOrEqualTo(n, precalculatedPrimes) {
    const primes = precalculatedPrimes ? precalculatedPrimes : []
    const maxPrime = primes.length ? primes[primes.length - 1] : 2
    for (let i = maxPrime; i <= n; i++) {
        let isPrime = true
        for (let p of primes) {
            isPrime = i % p !== 0
            if (!isPrime) break
        }
        if (isPrime) {
            primes.push(i)
        }
    }
    return primes
}

module.exports = primesSmallerOrEqualTo

function main() {
    const primes = primesSmallerOrEqualTo(2000000)
    console.log(primes.reduce((a, b) => a + b))
}

if (require.main === module) {
    main()
}